File: src/main/java/com/example/E_WardApplication/service/ChangeRequestService.java
package com.example.E_WardApplication.service;

import com.example.E_WardApplication.dto.ChangeRequestDTO;

import java.util.List;

public interface ChangeRequestService {

    ChangeRequestDTO create(ChangeRequestDTO dto);
    ChangeRequestDTO getById(Long id);
    List<ChangeRequestDTO> getAll();
    ChangeRequestDTO approve(Long id, Long approverUserId);
    ChangeRequestDTO reject(Long id, Long approverUserId);

}

File: e-ward-frontend/src/services/changeRequestService.js
import api from "./api";

function create(dto) {
  return api.post("/change-requests", dto);
}

function getAll() {
  return api.get("/change-requests");
}

function approve(id, approverUserId) {
  return api.post(`/change-requests/${id}/approve?approverUserId=${approverUserId}`);
}

function reject(id, approverUserId) {
  return api.post(`/change-requests/${id}/reject?approverUserId=${approverUserId}`);
}

export default { create, getAll, approve, reject };

